app/controllers/smart_proxies_controller.rb

```diff
@@ -49,6 +54,25 @@ def version
     render :json => {:success => true, :message => version[:message]}
   end
 
+  def plugin_version
+    plugin = params[:plugin]
+    begin
+      version = @smart_proxy.plugin_version(plugin)
+    rescue Foreman::Exception => exception
+      render :json => {:success => false, :message => exception.message} and return
+    end
+    render :json => {:success => true, :message => version[:message]}
+  end
+
+  def tftp_server
+    begin
+      tftp_server = @smart_proxy.tftp_server
+    rescue Foreman::Exception => exception
+      render :json => {:success => false, :message => exception.message} and return
+    end
+    render :json => {:success => true, :message => tftp_server[:message]}
```

Comment:
nitpick: it doesn't feel DRY, are you assuming the method would be different in the future? (or maybe its common infrastructure for every plugin author to add a section)?
you could do something like (not tested)

```ruby

def plugin_version
  plugin = params[:plugin]
  fetch_proxy do 
    response = @smart_proxy.plugin_version(plugin)
    render :json => response
  end
end

private

def fetch_proxy &block
  response = yield
  return {:success => true, :message => response[:message]}
rescue Foreman::Exception => exception
  return {:success => false, :message => exception.message}
end